clang/test/dpct/USM-restricted.cu

```diff
@@ -1095,3 +1095,29 @@ int foo17() {
   cudaMemcpy(d_in_data_2, h_data, mem_size, cudaMemcpyHostToDevice);
   return 0;
 }
+
+#define MAX_MASK_WIDTH 10
+
+template <typename T>
+__constant__ T mask[MAX_MASK_WIDTH];
+
+template <typename T>
+void foo18() {
+  size_t size_bytes = sizeof(double);
+  double *a = (T *)malloc(size_bytes);
+  double *d_a;
+  cudaMalloc((void **)&d_a, size_bytes);
+  // CHECK: q_ct1.memcpy(d_a, a, size_bytes);
+  // CHECK-NEXT: T h_mask[MAX_MASK_WIDTH];
+  // CHECK-NEXT: q_ct1.memcpy(mask.get_ptr(), h_mask, sizeof(double)).wait();
```

Comment:
Output/USM-restricted/USM-restricted.dp.cpp:1351:16: error: use of variable template 'mask' requires template arguments
 1351 |   q_ct1.memcpy(mask.get_ptr(), h_mask, sizeof(double)).wait();
      |                ^
Output/USM-restricted/USM-restricted.dp.cpp:1336:36: note: template is declared here
 1335 | template <typename T>
      | ~~~~~~~~~~~~~~~~~~~~~
 1336 | static dpct::constant_memory<T, 1> mask(MAX_MASK_WIDTH);
      |                                    ^